File: Programmers/큰 수 만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string number, int k) {
    string answer = "";
    
    int startIdx = 0;
    int maxIdx = 0;
    for (int i=0; i<number.size() - k; i++) {
        char max = number[startIdx];
        
        for (int j=startIdx; j<=k+i; j++) {
            if (number[j] > max) {
                max = number[j];
                maxIdx = j;
            }
        }
        
        startIdx = maxIdx + 1;
        maxIdx = startIdx;
        answer += max;
    }
    
    return answer;
}

File: 프로그래머스/큰 수 만들기.py
def solution(number, k):
    number_list = list(number)
    stack = []
    for number in number_list:
        while stack and stack[-1] < number and k > 0:
            stack.pop()
            k -= 1
        stack.append(number)
    if k > 0:
        stack = stack[:-k]
    answer = ''.join(stack)
    return answer